app/Models/User.php

```diff
@@ -1118,49 +1076,62 @@ public function defaultGroup(): Group
         return $groups->byId($this->group_id) ?? $groups->byIdentifier('default');
     }
 
-    public function groupIds()
+    /**
+     * Get the group IDs of the user's active usergroups.
+     *
+     * @return array<int, int>
+     */
+    public function groupIds(): array
     {
         return $this->memoize(__FUNCTION__, function () {
-            $ret = [
-                'active' => [],
-                'pending' => [],
-            ];
-
-            foreach ($this->userGroups as $userGroup) {
-                $key = $userGroup->user_pending ? 'pending' : 'active';
-                $ret[$key][] = $userGroup->group_id;
-            }
-
-            return $ret;
+            return $this
+                ->userGroups
+                ->where('user_pending', false)
+                ->pluck('group_id')
+                ->all();
         });
     }
 
-
-    public function findUserGroup($group, bool $activeOnly): ?UserGroup
+    /**
+     * Get the user's usergroup corresponding to a specified group.
+     */
+    public function findUserGroup(Group|string $group, bool $includePending = false): ?UserGroup
```

Comment:
I'm not keen on this type of param type overloading and having conditionals in the body to convert it when the specific type is known before hand; this isn't a parsing function.

It also means `findUserGroup` implicitly picks up the auto group creating behaviour of `Groups::byIdentifier` as an unexpected side-effect.